### PR TITLE
scale-mesh demo

### PR DESCRIPTION
**Pull request by jmazzitelli**
This provides a sophisticated way to generate large graphs with different mesh characteristics.

See the README.adoc for the different mesh types this can produce (the README provides images of the app graphs for the different mesh types so you can see what each mesh type will generally look like).

You do not need to clone this repo to be able to run this. The README provides the commands to run if you do not have this repo cloned - you simply use the `make` argument `-f <(curl -L https://raw.githubusercontent.com/kiali/demos/master/scale-mesh/Makefile)`:

```
make install -e MESH_TYPE=breadth -f <(curl -L https://raw.githubusercontent.com/kiali/demos/master/scale-mesh/Makefile)
```

If you do have this repo cloned, you do not need to pass that `-f` option on the `make` command.

To test this for all the different mesh types, you simply have to run these commands (by default, they will each go in their own single namespace - install all of these, then go to your Kiali graph and select the different namespaces to see the different graphs):

```
make install -e MESH_TYPE=breadth
make install -e MESH_TYPE=breadth-sink
make install -e MESH_TYPE=circle
make install -e MESH_TYPE=circle-callback
make install -e MESH_TYPE=depth
make install -e MESH_TYPE=depth-sink
make install -e MESH_TYPE=hourglass
```

### CODE REVIEW

File: scale-mesh/README.adoc [hunk 1]
```diff
@@ -0,0 +1,92 @@
+:imagesdir: images
+
+# Scale Mesh
+
+The scale mesh demo can be used for perfomance and scalability testing as well as for exercising Kiali's graph functionality.
```
Comment (jshaughn) [inline scale-mesh/README.adoc:5]: perfomance --> performance
testing as --> testing, as
functionality --> feature

**Review by jshaughn — APPROVED** [review]
Approving but before merging please address the single comment for the README

File: scale-mesh/README.adoc (hunk 1 above)
Comment (jmazzitelli) [reply scale-mesh/README.adoc]: fine. done.

**Comment by jmazzitelli** [comment]
One thing I could have done (and didn't think of it until after I finished all of this) was I really didn't need a Makefile. I could have written this with a bash script that accepts some cmdline options.

If we do not want to require the user to have "make" I can always convert this to just use a bash script (I don't think it will be hard to convert). But for now, I'll merge it as is. We can convert the make stuff later in a future PR if that's what people would prefer.

**Pull request merged**